doc/dev/osd_internals/snaps.rst

```diff
@@ -126,3 +126,84 @@ up to 8 prefixes need to be checked to determine all hobjects in a particular
 snap for a particular PG.  Upon split, the prefixes to check on the parent
 are adjusted such that only the objects remaining in the PG will be visible.
 The children will immediately have the correct mapping.
+
+clone_overlap
+-------------
+Each SnapSet attached to the head object contains the overlapping intervals
+between clones objects for optimising the space usage.
+The overlapping intervals are stored within the ``clone_overlap`` map, each element in the
+map stores the snap id and the corresponding overlap with the next newest clone.
+
+See the following example using a 4 Byte object:
+
+object content is: [AAAA]
+
+listnsnap output is as follows:
++------------+------------+-----------+-------------------------+
+| cloneid    | snaps      | size      | overlap                 |
++============+============+===========+=========================+
+| head       | -          | 4         |                         |
++------------+------------+-----------+-------------------------|
+
+After taking a snapshot (id 1) and re-writing the first 2 Bytes of the object,
+the clone created will overlap with the new head object in it's last 2 Bytes.
```

Comment:
s/it's/its/

I will handle this in a future grammar-and-punctuation-correction commit.

Reply:
Ah, thanks! I'd prefer waiting for an additional approval to verify that the logic described adds up.